### PR TITLE
Use new `if let` syntax wherever possible. Fixes #4153.

### LINKED ISSUE #4153: Ungate and use `if let` everywhere
Similar to https://github.com/rust-lang/rust/pull/19405/files

Use `if let` to reduce match statements of the form 

```
match foo {
 (some pattern here) => {....}
 _ => ()

}
```

with 

```
if let (pattern) = foo {
...
}
```

(This can be handled as smaller easy bugs, one for each crate)

We also might want to consider replacing `Option::map` with an `if let`; `map` is slightly inefficient  (though perhaps it will become better post unboxed closures)

### PR DESCRIPTION
**Pull request by zslayton**
Opening this PR to invite feedback. 

Of the many `match` statement candidates for conversion to `if let`, several included `if` guards. Since `if let` doesn't support this syntax, I used nested if statements. If this is undesirable, say the word and I can revert those cases to `match`.

### CODE REVIEW

**Comment by Ms2ger** [comment]
@zslayton: thanks for your contribution! I reviewed on critic; it looks mostly good, but please push a fixup commit to revert the changes I marked.

**Comment by zslayton** [comment]
Thanks, @Ms2ger! I'll make those changes this evening and re-push.

**Comment by Ms2ger** [comment]
Thank you! Please squash the two commits now, and we'll merge.

**Comment by zslayton** [comment]
Squashed!

**Comment by Ms2ger** [comment]
Thanks!

File: components/script/dom/htmloptgroupelement.rs [hunk 1]
```diff
@@ -75,14 +74,13 @@ impl<'a> VirtualMethods for JSRef<'a, HTMLOptGroupElement> {
                     child.set_enabled_state(false);
                 }
             },
-            _ => ()
+            _ => (),
```
Comment (bfrohs) [inline components/script/dom/htmloptgroupelement.rs]: Added by accident?

Comment (zslayton) [inline components/script/dom/htmloptgroupelement.rs]: Ah, yes. This was unintentional. @Ms2ger, would you like me to make another PR to remove this stray comma?

Comment (Ms2ger) [inline components/script/dom/htmloptgroupelement.rs]: Nah, I actually like this more.
